### PR TITLE
spread: add unit suite

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -283,4 +283,27 @@ suites:
             fi
             $TESTSLIB/reset.sh
 
+    tests/unit/:
+        summary: Tests involving binary building processes
```
Comment (pedronis) [inline spread.yaml]: Suite to run non-go unit tests and go unit tests for other runtimes

Comment (fgimenez) [inline spread.yaml]: @pedronis done thx!

**Comment by zyga** [comment]
I think you need to patch the suite to install the core snap. Now that it is a suite by itself it seems to be failing on this (repeatedly). It looks exactly like the error we sometimes hit in the main suite but now, since this is the only test, it happens all the time.

File: spread.yaml [hunk 2]
```diff
@@ -277,11 +277,27 @@ suites:
                 exit 0
             fi
             apt-get purge -y snapd snap-confine
-        restore-each: |
```
Comment (pedronis) [inline spread.yaml]: I think we want to keep this one, no?

Comment (fgimenez) [inline spread.yaml]: Done, thx!

**Pull request merged**